### PR TITLE
added option to override messagelist notifications

### PR DESCRIPTION
**Pull request by ambaldwin**
# Submit a pull request

## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] The code changes follow best practices
- [ ] Code changes are tested (add some information if not applicable)

### CODE REVIEW

File: docusaurus/docs/React/contexts/component-context.mdx [hunk 1]
```diff
@@ -171,6 +171,14 @@ Custom UI component for a deleted message.
...
 | component | [MessageDeleted](https://github.com/GetStream/stream-chat-react/blob/master/src/components/Message/MessageDeleted.tsx) |
 
+### MessageListNotifications
+
+Custom UI component that displays message and connection status notifications in the `MessageList`.
+
+| Type      | Default                                                                                                                                  |
+| --------- | ---------------------------------------------------------------------------------------------------------------------------------------- |
+| component | [DefaultMessageListNotifications](https://github.com/GetStream/stream-chat-react/blob/master/src/components/MessageList/MessageList.tsx) |
+
```
Comment (DanC5) [inline docusaurus/docs/React/contexts/component-context.mdx:181]: why the spacing look weird on the borders in the mdx files?

File: docusaurus/docs/React/custom-code-examples/adding-notification.mdx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import CustomNotification from '../assets/CustomNotification.png';
+
+In this example, we demonstrate how to use the method `addNotification` pulled from the [`ChannelActionContext`](../contexts/channel_action_context#addNotification) in order to add a custom notification message to the `MessageList` at our desired time specification.
+
+## addNotification Method
+
+This method receives two arguments: text and type. The text parameter is the notification text, and the type argument is of type string and is either 'success' or 'error'. The type determines the style added.
+These messages are temporary in the list and are removed after five seconds.
+
+This method is used extensively in the library by handler functions to notify of success or failure and are usually used in conjunction with Javascript event listeners or API event listeners.
```
Comment (DanC5) [inline docusaurus/docs/React/custom-code-examples/adding-notification.mdx:16]: JavaScript ;)

File: docusaurus/docs/React/custom-code-examples/adding-notification.mdx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+Since `sendMessage` is drawn from the `ChannelActionContext`, we must create an inner component that is a child of `Channel` and
+call the `useChannelActionContext` custom hook.
+:::
+
+## The Code
+
+```jsx
+  const ChannelInner = () => {
+    const { addNotification } = useChannelActionContext();
+    const { channel } = useChatContext();
```
Comment (DanC5) [inline docusaurus/docs/React/custom-code-examples/adding-notification.mdx:38]: channel should be pulled from useChannelStateContext, then won't need to optional chain below

File: docusaurus/docs/React/custom-code-examples/system-message.mdx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        <div className='custom-system-message'>
+            <span>Event: {text} at {date}</span>
+            <span>
+                Actor: <Avatar image={user?.image} shape={'square'} size={20} />
+                {user?.name}
+            </span>
+        </div>
+    )
+}
+return null;
```
Comment (DanC5) [inline docusaurus/docs/React/custom-code-examples/system-message.mdx:67]: let's newline here just to let it breathe!

**Review by DanC5 — APPROVED** [review]

File: docusaurus/docs/React/contexts/component-context.mdx (hunk 1 above)
Comment (ambaldwin) [reply docusaurus/docs/React/contexts/component-context.mdx:181]: I don't know, but it looks fine when running and also doesn't seem to have spaces in the file...

**Pull request merged**